storage/statedb/cache_redis_test.go

```diff
@@ -98,22 +99,68 @@ func TestNewRedisCache(t *testing.T) {
 	assert.Equal(t, bytes.Compare(value, hasValue), 0)
 }
 
-// TestNewRedisCache_Set_LargeData check whether redis cache can store an large data (5MB).
-func TestNewRedisCache_Set_LargeData(t *testing.T) {
+// TestRedisCache_Set_LargeData check whether redis cache can store an large data (5MB).
+func TestRedisCache_Set_LargeData(t *testing.T) {
 	cache, err := NewRedisCache(getTestRedisConfig())
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	key, value := randBytes(32), randBytes(5*1024*1024) // 5MB value
 	cache.Set(key, value)
+	time.Sleep(100 * time.Millisecond)
 
 	retValue := cache.Get(key)
 	assert.Equal(t, bytes.Compare(value, retValue), 0)
 }
 
-// TestNewRedisCache_Timeout test timout feature of redis client.
-func TestNewRedisCache_Timeout(t *testing.T) {
+func TestRedisCache_Set_LargeNumberItems(t *testing.T) {
```

Comment:
When the Set channel is full, not writing is expected behavior.
I am not sure what this test is trying to do.
Is this an example code for asynchronous write without failure?

Reply:
As you mentioned, it is hard to test channel full because it's expected behavior. The test just ensures that all items passed to the channel are consumed correctly (not remained in the channel) in normal operation.